crates/bevy_ecs/src/system/query.rs

```diff
@@ -692,14 +724,20 @@ impl<'w, 's, Q: WorldQuery, F: WorldQuery> Query<'w, 's, Q, F> {
         };
     }
 
-    /// Runs `f` on each query result in parallel using the [`World`]'s [`ComputeTaskPool`].
-    /// See [`Self::par_for_each`] for more details.
+    /// Runs `f` on each read-only query item in parallel.
+    ///
+    /// Parallelization is achieved by using the [`World`]'s [`ComputeTaskPool`].
     ///
     /// # Panics
-    /// The [`ComputeTaskPool`] is not initialized. If using this from a query that is being
-    /// initialized and run from the ECS scheduler, this should never panic.
+    ///
+    /// This method panics if the [`ComputeTaskPool`] resource is added to the `World` before using this method.
+    /// If using this from a query that is being initialized and run from the [`Schedule`](crate::schedule::Schedule), this should never panic.
```

Comment:
Same as above, all or nothing :)

```suggestion
    /// If using this from a query that is being initialized and run from the [`Schedule`](crate::schedule::Schedule), this never panics.
```